org/enigma/TargetHandler.java

```diff
@@ -271,30 +280,44 @@ public int compareTo(Combo c)
 			{
 			String ey = file.getName();
 			if (!ey.endsWith(".ey")) continue; //$NON-NLS-1$
+			TargetSelection ps = null;
 			try
 				{
 				YamlNode node = YamlParser.parse(file);
-				YamlNode exeVarsNode = (YamlNode) node.getM("EXE-Vars");
-
-				TargetSelection ps = new TargetSelection();
+				ps = new TargetSelection();
 				ps.id = ey.substring(0,ey.length() - 3);
 				ps.name = node.getMC("Name"); //$NON-NLS-1$
 				ps.desc = node.getMC("Description",null); //$NON-NLS-1$
 				ps.auth = node.getMC("Maintainer",null); //$NON-NLS-1$
-				ps.ext = exeVarsNode.getMC("Build-Extension",null); //$NON-NLS-1$
-				ps.outputexe = exeVarsNode.getMC("Run-output",null); //$NON-NLS-1$
 				ps.depends = new HashMap<String,Set<String>>();
 				Set<String> target = new HashSet<String>();
 				String targplat = node.getMC("Target-platform",null); //$NON-NLS-1$
 				if (target != null) targplat = normalize(targplat);
 				target.add(targplat);
 				ps.depends.put("target",target); //$NON-NLS-1$
 				if (node.getBool("Native",false)) defCompiler = ps; //$NON-NLS-1$
-				tCompilers.add(ps);
+				try
+					{
+					YamlNode exeVarsNode = (YamlNode) node.getM("EXE-Vars"); //$NON-NLS-1$
+					ps.ext = exeVarsNode.getMC("Build-Extension",null); //$NON-NLS-1$
+					ps.outputexe = exeVarsNode.getMC("Run-output",null); //$NON-NLS-1$
+					}
+				catch (IndexOutOfBoundsException e)
+					{ // there was no "EXE-Vars" key, which is optional
+					ps.ext = null;
```

Comment:
Try the old one, please. I can't run bisection searches to catch regressions if LGM supports only one revision and onward.

Reply:
Alright, I've done so:

In this case, whether the keys appear under the old location or not, the user is never presented with any error. Consider the following where the keys do not exist under the old or new location.
![Keys Missing Under Old and New Location](https://user-images.githubusercontent.com/3212801/41804797-a1a0066a-766a-11e8-9b99-be9c3677b126.png)

Reply:
ENIGMA will never be in a mixed state, because all ENIGMA's source is local to the ENIGMA repo. I asked you to make this change because LGM exists outside of ENIGMA, so checking out a branch at HEAD~50 would require the version of LGM before this change, while checking out at HEAD will require a version of LGM including this change. I don't expect people to revert only their configs and not rebuild ENIGMA.

Reply:
Makes sense, that's what I thought you were going for.